cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/KubernetesVersion.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2017-2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.operator;
+
+/**
+ * Represents Kubernetes version which CO runs on
+ */
+public class KubernetesVersion implements Comparable<KubernetesVersion> {
+
+    private int major;
+    private int minor;
+    public static final KubernetesVersion V1_8 = new KubernetesVersion(1, 8);
+    public static final KubernetesVersion V1_9 = new KubernetesVersion(1, 9);
+    public static final KubernetesVersion V1_10 = new KubernetesVersion(1, 10);
+    public static final KubernetesVersion V1_11 = new KubernetesVersion(1, 11);
+    public static final KubernetesVersion V1_12 = new KubernetesVersion(1, 12);
```

Comment:
There is Kubernetes 1.13 and 1.14 available already, so you should probably add them. I also don't really like this list which will get unmaintained soon. But since it seems you can easily create your own versions I can probably live with it.

Reply:
Really these constants exist only so (at the appropriate places) we can write stuff like:

```
if (actualVersion.compare(V1_11) > 0) {
    // ...
}
```

If we don't need to write stuff like that in the production code then we should remove that particular constant.

Reply:
I can live with that, but I guess it could be still a bit confusing why we have just some versions listed there. Some comment might help.